### PR TITLE
instrumentation tests are executed with BUCK

### PR DESCRIPTION
**Pull request by bestander**
Added ability to run instrumentation tests with BUCK.

This change uses BUCK to build and run instrumentation tests facebook style.
The gains are that we can execute the same tests internally at FB and in OSS.
Also running tests not via graddle:connect command is 1.5 minutes faster.

I'll keep keep an eye on stability Gradle and BUCK builds for a while.

### CODE REVIEW

**Comment by bestander** [comment]
Currently I get these results:
```
$ buck test ReactAndroid/src/androidTest/buck-runner:react-oss-tests
[-] PROCESSING BUCK FILES...FINISHED 0.2s [100%]
[-] DOWNLOADING... (0.00 B/S AVG, TOTAL: 0.00 B, 0 Artifacts)
[-] BUILDING...FINISHED 3.7s [100%] (174/324 JOBS, 5 UPDATED, 1.5% CACHE MISS)
[-] TESTING...FINISHED 1.7s
RESULTS FOR //ReactAndroid/src/androidTest/buck-runner:react-oss-tests
NOTESTS <100ms  0 Passed   0 Skipped   0 Failed   com.facebook.react.tests (192.168.56.101:5555)
TESTS PASSED
```

**Comment by bestander** [comment]
To execute locally:
```
./gradlew :ReactAndroid:packageReactNdkLibsForBuck
node local-cli/cli.js bundle --platform android --dev true --entry-file ReactAndroid/src/androidTest/assets/TestBundle.js --bundle-output ReactAndroid/src/androidTest/assets/AndroidTestBundle.js
buck test ReactAndroid/src/androidTest/java/com/facebook/react/tests
```

File: ReactAndroid/src/androidTest/BUCK_temp [hunk 1]
```diff
@@ -1,66 +0,0 @@
...
-  name = 'react_oss_test',
-  apk = ':react_oss_test_apk',
-)
-
-sh_binary(
-  name = 'integration_test_oss_bundle_js',
-  main = 'buildBundle.sh',
-#     deps = [':InstallReactNativeNodeModules'],
-  visibility = ['PUBLIC'],
-)
```
Comment (mkonicek) [inline ReactAndroid/src/androidTest/BUCK_temp]: cool :)

File: scripts/run-android-instrumentation-tests.sh [hunk 2]
```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+export PATH="$ANDROID_HOME/platform-tools:$ANDROID_HOME/tools:$PATH"
+
+# clear the logs
+adb logcat -c
+
+# run tests and check output
+python - $1 << END
```
Comment (mkonicek) [inline scripts/run-android-instrumentation-tests.sh]: Whoa didn't know you could write Python in a .sh file! Nice :)

File: scripts/run-android-instrumentation-tests.sh [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+done = True
+print stderr
+print stdout
+
+if success.search(stderr + stdout):
+  sys.exit(0)
+else:
+  # dump the logs
+  sp.Popen(['adb', 'logcat', '-d']).communicate()
```
Comment (mkonicek) [inline scripts/run-android-instrumentation-tests.sh]: Super cool!

**Comment by mkonicek** [comment]
Really cool!

@facebook-github-bot shipit

**Comment by mkonicek** [comment]
@bestander Failed to land with:

    BUILD FAILED: 'Libraries/FBReactKit/js/react-native-github/ReactAndroid/src/androidTest/assets/ScrollViewTestModule.js'
    in '//Libraries/FBReactKit:GroupsJS' crosses a buck package boundary.
    This file is owned by
    'Libraries/FBReactKit/js/react-native-github/ReactAndroid/src/androidTest/assets'.
    Find the owning rule that references
    'Libraries/FBReactKit/js/react-native-github/ReactAndroid/src/androidTest/assets/ScrollViewTestModule.js',
    and use a reference to that rule instead of referencing the desired file directly.

Check out Sandastle.

**Comment by bestander** [comment]
yep, working on it

**Pull request closed without merging**